build.gradle.kts

```diff
@@ -140,77 +140,62 @@ allprojects {
-                // only execute task if the opentelemetry agent does not exist. invoke the "clean" task to force
-                onlyIf {
-                    !agentFile.exists()
-                }
-                // this task could be the first in the graph, so "build/" may not yet exist. Let's be defensive
-                doFirst {
-                    buildDir.mkdirs()
-                }
-                // download the jar file
-                doLast {
-                    val download = { url: String, destFile: File ->
-                        ant.invokeMethod(
-                            "get",
-                            mapOf("src" to url, "dest" to destFile)
-                        )
-                    }
-                    logger.lifecycle("Downloading OpenTelemetry Agent")
-                    download(openTelemetryAgentUrl, agentFile)
+            val downloadOpentelemetryAgent = tasks.create("downloadOpentelemetryAgent", Copy::class) {
+                val openTelemetry = configurations.create("open-telemetry")
+
+                dependencies {
+                    openTelemetry(libs.opentelemetry.javaagent)
                 }
-            }
 
-            val shadowJarTask = tasks.named(ShadowJavaPlugin.SHADOW_JAR_TASK_NAME).get()
+                from(openTelemetry)
+                into("build/resources/otel")
+                rename { "opentelemetry-javaagent.jar" }
+            }
 
-            // this task copies some legal docs into the build folder, so we can easily copy them into the docker images
             val copyLegalDocs = tasks.create("copyLegalDocs", Copy::class) {
                 from(project.rootProject.projectDir)
-                into("${buildDir}/legal")
+                into("build/legal")
                 include("SECURITY.md", "NOTICE.md", "DEPENDENCIES", "LICENSE")
-                dependsOn(shadowJarTask)
             }
 
             val copyDockerfile = tasks.create("copyDockerfile", Copy::class) {
                 from(rootProject.projectDir.toPath().resolve("resources"))
                 into(project.layout.buildDirectory.dir("resources").get().dir("docker"))
                 include("Dockerfile")
             }
-            shadowJarTask.dependsOn(copyDockerfile)
+
+            val shadowJarTask = tasks.named(ShadowJavaPlugin.SHADOW_JAR_TASK_NAME).get()
+
+            shadowJarTask
+                .dependsOn(copyDockerfile)
+                .dependsOn(copyLegalDocs)
+                .dependsOn(downloadOpentelemetryAgent)
 
             //actually apply the plugin to the (sub-)project
             apply(plugin = "com.bmuschko.docker-remote-api")
 
             val dockerTask: DockerBuildImage = tasks.create("dockerize", DockerBuildImage::class) {
-                dockerFile.set(project.layout.buildDirectory.asFile.get().toPath().resolve("resources").resolve("docker").resolve("Dockerfile").toFile())
+                dockerFile.set(File("build/resources/docker/Dockerfile"))
```

Comment:
this will likely cause problems on windows. I know we don't officially support it, but must we intentionally break it?

Reply:
in any case the buiild-args passed at line 192-194 do specify linux paths. I can switch all of them, not really sure about the value given

Reply:
yeah, true.